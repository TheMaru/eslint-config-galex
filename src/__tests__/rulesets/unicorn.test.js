const { createUnicornRules } = require('../../rulesets/unicorn');

describe('createUnicornRules', () => {
  test('matches snapshot without typescript', () => {
    const project = {
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createUnicornRules(project)).toMatchSnapshot();
  });

  test('matches snapshot with typescript', () => {
    const project = {
      typescript: {
        hasTypeScript: true,
      },
    };

    expect(createUnicornRules(project)).toMatchSnapshot();
  });

  test('allows passing custom rules', () => {
    const rule = 'unicorn/better-regex';
    const level = 'off';

    const project = {
      customRules: {
        [rule]: level,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    const defaultRuleset = createUnicornRules({
      ...project,
      customRules: {},
    });

    const result = createUnicornRules(project);

    expect(result[rule]).toBe(level);
    expect(result[rule]).not.toBe(defaultRuleset[rule]);

    expect(result).toMatchSnapshot();
  });
});
