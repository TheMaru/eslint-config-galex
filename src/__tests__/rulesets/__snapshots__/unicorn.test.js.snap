// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUnicornRules allows passing custom rules 1`] = `
Object {
  "unicorn/better-regex": "off",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-destructuring": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/empty-brace-spaces": "off",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/import-style": "off",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-callback-reference": "off",
  "unicorn/no-array-for-each": "off",
  "unicorn/no-array-method-this-argument": "warn",
  "unicorn/no-array-push-push": "warn",
  "unicorn/no-array-reduce": "off",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-document-cookie": "warn",
  "unicorn/no-for-loop": "off",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-instanceof-array": "error",
  "unicorn/no-keyword-prefix": "off",
  "unicorn/no-lonely-if": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-array": "error",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-static-only-class": "error",
  "unicorn/no-this-assignment": "warn",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-length-check": "error",
  "unicorn/no-useless-spread": "error",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/numeric-separators-style": "warn",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-array-flat": "warn",
  "unicorn/prefer-array-flat-map": "error",
  "unicorn/prefer-array-index-of": "warn",
  "unicorn/prefer-array-some": "warn",
  "unicorn/prefer-at": "off",
  "unicorn/prefer-date-now": "warn",
  "unicorn/prefer-default-parameters": "warn",
  "unicorn/prefer-dom-node-append": "error",
  "unicorn/prefer-dom-node-dataset": "error",
  "unicorn/prefer-dom-node-remove": "error",
  "unicorn/prefer-dom-node-text-content": "error",
  "unicorn/prefer-includes": "off",
  "unicorn/prefer-keyboard-event-key": "error",
  "unicorn/prefer-math-trunc": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-module": "off",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-protocol": "off",
  "unicorn/prefer-number-properties": Array [
    "warn",
    Object {
      "checkInfinity": false,
    },
  ],
  "unicorn/prefer-object-from-entries": "error",
  "unicorn/prefer-object-has-own": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-prototype-methods": "warn",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-regexp-test": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-string-replace-all": "error",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-string-starts-ends-with": "off",
  "unicorn/prefer-string-trim-start-end": "error",
  "unicorn/prefer-switch": Array [
    "error",
    Object {
      "emptyDefaultCase": "no-default-comment",
    },
  ],
  "unicorn/prefer-ternary": "off",
  "unicorn/prefer-top-level-await": "warn",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/require-array-join-separator": "warn",
  "unicorn/require-number-to-fixed-digits-argument": "warn",
  "unicorn/require-post-message-target-origin": "error",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules enables esnext rules given react 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-destructuring": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/empty-brace-spaces": "off",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/import-style": "off",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-callback-reference": "off",
  "unicorn/no-array-for-each": "off",
  "unicorn/no-array-method-this-argument": "warn",
  "unicorn/no-array-push-push": "warn",
  "unicorn/no-array-reduce": "off",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-document-cookie": "warn",
  "unicorn/no-for-loop": "warn",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-instanceof-array": "error",
  "unicorn/no-keyword-prefix": "off",
  "unicorn/no-lonely-if": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-array": "error",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-static-only-class": "error",
  "unicorn/no-this-assignment": "warn",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-length-check": "error",
  "unicorn/no-useless-spread": "error",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/numeric-separators-style": "warn",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-array-flat": "warn",
  "unicorn/prefer-array-flat-map": "error",
  "unicorn/prefer-array-index-of": "warn",
  "unicorn/prefer-array-some": "warn",
  "unicorn/prefer-at": "off",
  "unicorn/prefer-date-now": "warn",
  "unicorn/prefer-default-parameters": "warn",
  "unicorn/prefer-dom-node-append": "error",
  "unicorn/prefer-dom-node-dataset": "error",
  "unicorn/prefer-dom-node-remove": "error",
  "unicorn/prefer-dom-node-text-content": "error",
  "unicorn/prefer-includes": "warn",
  "unicorn/prefer-keyboard-event-key": "error",
  "unicorn/prefer-math-trunc": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-module": "off",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-protocol": "off",
  "unicorn/prefer-number-properties": Array [
    "warn",
    Object {
      "checkInfinity": false,
    },
  ],
  "unicorn/prefer-object-from-entries": "error",
  "unicorn/prefer-object-has-own": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-prototype-methods": "warn",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-regexp-test": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-string-replace-all": "error",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-string-starts-ends-with": "error",
  "unicorn/prefer-string-trim-start-end": "error",
  "unicorn/prefer-switch": Array [
    "error",
    Object {
      "emptyDefaultCase": "no-default-comment",
    },
  ],
  "unicorn/prefer-ternary": "off",
  "unicorn/prefer-top-level-await": "warn",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/require-array-join-separator": "warn",
  "unicorn/require-number-to-fixed-digits-argument": "warn",
  "unicorn/require-post-message-target-origin": "error",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules enables esnext rules given typsecript with certain lib criteria 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-destructuring": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/empty-brace-spaces": "off",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/import-style": "off",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-callback-reference": "off",
  "unicorn/no-array-for-each": "off",
  "unicorn/no-array-method-this-argument": "warn",
  "unicorn/no-array-push-push": "warn",
  "unicorn/no-array-reduce": "off",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-document-cookie": "warn",
  "unicorn/no-for-loop": "off",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-instanceof-array": "error",
  "unicorn/no-keyword-prefix": "warn",
  "unicorn/no-lonely-if": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-array": "error",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-static-only-class": "error",
  "unicorn/no-this-assignment": "warn",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-length-check": "error",
  "unicorn/no-useless-spread": "error",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/numeric-separators-style": "warn",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-array-flat": "warn",
  "unicorn/prefer-array-flat-map": "error",
  "unicorn/prefer-array-index-of": "warn",
  "unicorn/prefer-array-some": "warn",
  "unicorn/prefer-at": "off",
  "unicorn/prefer-date-now": "warn",
  "unicorn/prefer-default-parameters": "warn",
  "unicorn/prefer-dom-node-append": "error",
  "unicorn/prefer-dom-node-dataset": "error",
  "unicorn/prefer-dom-node-remove": "error",
  "unicorn/prefer-dom-node-text-content": "error",
  "unicorn/prefer-includes": "off",
  "unicorn/prefer-keyboard-event-key": "error",
  "unicorn/prefer-math-trunc": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-module": "off",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-protocol": "off",
  "unicorn/prefer-number-properties": Array [
    "warn",
    Object {
      "checkInfinity": false,
    },
  ],
  "unicorn/prefer-object-from-entries": "error",
  "unicorn/prefer-object-has-own": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-prototype-methods": "warn",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-regexp-test": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-string-replace-all": "error",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-string-starts-ends-with": "off",
  "unicorn/prefer-string-trim-start-end": "error",
  "unicorn/prefer-switch": Array [
    "error",
    Object {
      "emptyDefaultCase": "no-default-comment",
    },
  ],
  "unicorn/prefer-ternary": "off",
  "unicorn/prefer-top-level-await": "warn",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/require-array-join-separator": "warn",
  "unicorn/require-number-to-fixed-digits-argument": "warn",
  "unicorn/require-post-message-target-origin": "error",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot with react, without typescript 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-destructuring": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/empty-brace-spaces": "off",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/import-style": "off",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-callback-reference": "off",
  "unicorn/no-array-for-each": "off",
  "unicorn/no-array-method-this-argument": "warn",
  "unicorn/no-array-push-push": "warn",
  "unicorn/no-array-reduce": "off",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-document-cookie": "warn",
  "unicorn/no-for-loop": "warn",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-instanceof-array": "error",
  "unicorn/no-keyword-prefix": "off",
  "unicorn/no-lonely-if": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-array": "error",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-static-only-class": "error",
  "unicorn/no-this-assignment": "warn",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-length-check": "error",
  "unicorn/no-useless-spread": "error",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/numeric-separators-style": "warn",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-array-flat": "warn",
  "unicorn/prefer-array-flat-map": "error",
  "unicorn/prefer-array-index-of": "warn",
  "unicorn/prefer-array-some": "warn",
  "unicorn/prefer-at": "off",
  "unicorn/prefer-date-now": "warn",
  "unicorn/prefer-default-parameters": "warn",
  "unicorn/prefer-dom-node-append": "error",
  "unicorn/prefer-dom-node-dataset": "error",
  "unicorn/prefer-dom-node-remove": "error",
  "unicorn/prefer-dom-node-text-content": "error",
  "unicorn/prefer-includes": "warn",
  "unicorn/prefer-keyboard-event-key": "error",
  "unicorn/prefer-math-trunc": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-module": "off",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-protocol": "off",
  "unicorn/prefer-number-properties": Array [
    "warn",
    Object {
      "checkInfinity": false,
    },
  ],
  "unicorn/prefer-object-from-entries": "error",
  "unicorn/prefer-object-has-own": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-prototype-methods": "warn",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-regexp-test": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-string-replace-all": "error",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-string-starts-ends-with": "error",
  "unicorn/prefer-string-trim-start-end": "error",
  "unicorn/prefer-switch": Array [
    "error",
    Object {
      "emptyDefaultCase": "no-default-comment",
    },
  ],
  "unicorn/prefer-ternary": "off",
  "unicorn/prefer-top-level-await": "warn",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/require-array-join-separator": "warn",
  "unicorn/require-number-to-fixed-digits-argument": "warn",
  "unicorn/require-post-message-target-origin": "error",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot with typescript & react 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-destructuring": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/empty-brace-spaces": "off",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/import-style": "off",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-callback-reference": "off",
  "unicorn/no-array-for-each": "off",
  "unicorn/no-array-method-this-argument": "warn",
  "unicorn/no-array-push-push": "warn",
  "unicorn/no-array-reduce": "off",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-document-cookie": "warn",
  "unicorn/no-for-loop": "off",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-instanceof-array": "error",
  "unicorn/no-keyword-prefix": "off",
  "unicorn/no-lonely-if": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-array": "error",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-static-only-class": "error",
  "unicorn/no-this-assignment": "warn",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-length-check": "error",
  "unicorn/no-useless-spread": "error",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/numeric-separators-style": "warn",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-array-flat": "warn",
  "unicorn/prefer-array-flat-map": "error",
  "unicorn/prefer-array-index-of": "warn",
  "unicorn/prefer-array-some": "warn",
  "unicorn/prefer-at": "off",
  "unicorn/prefer-date-now": "warn",
  "unicorn/prefer-default-parameters": "warn",
  "unicorn/prefer-dom-node-append": "error",
  "unicorn/prefer-dom-node-dataset": "error",
  "unicorn/prefer-dom-node-remove": "error",
  "unicorn/prefer-dom-node-text-content": "error",
  "unicorn/prefer-includes": "off",
  "unicorn/prefer-keyboard-event-key": "error",
  "unicorn/prefer-math-trunc": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-module": "off",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-protocol": "off",
  "unicorn/prefer-number-properties": Array [
    "warn",
    Object {
      "checkInfinity": false,
    },
  ],
  "unicorn/prefer-object-from-entries": "error",
  "unicorn/prefer-object-has-own": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-prototype-methods": "warn",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-regexp-test": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-string-replace-all": "error",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-string-starts-ends-with": "off",
  "unicorn/prefer-string-trim-start-end": "error",
  "unicorn/prefer-switch": Array [
    "error",
    Object {
      "emptyDefaultCase": "no-default-comment",
    },
  ],
  "unicorn/prefer-ternary": "off",
  "unicorn/prefer-top-level-await": "warn",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/require-array-join-separator": "warn",
  "unicorn/require-number-to-fixed-digits-argument": "warn",
  "unicorn/require-post-message-target-origin": "error",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot with typescript, without react 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-destructuring": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/empty-brace-spaces": "off",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/import-style": "off",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-callback-reference": "off",
  "unicorn/no-array-for-each": "off",
  "unicorn/no-array-method-this-argument": "warn",
  "unicorn/no-array-push-push": "warn",
  "unicorn/no-array-reduce": "off",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-document-cookie": "warn",
  "unicorn/no-for-loop": "off",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-instanceof-array": "error",
  "unicorn/no-keyword-prefix": "warn",
  "unicorn/no-lonely-if": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-array": "error",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-static-only-class": "error",
  "unicorn/no-this-assignment": "warn",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-length-check": "error",
  "unicorn/no-useless-spread": "error",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/numeric-separators-style": "warn",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-array-flat": "warn",
  "unicorn/prefer-array-flat-map": "error",
  "unicorn/prefer-array-index-of": "warn",
  "unicorn/prefer-array-some": "warn",
  "unicorn/prefer-at": "off",
  "unicorn/prefer-date-now": "warn",
  "unicorn/prefer-default-parameters": "warn",
  "unicorn/prefer-dom-node-append": "error",
  "unicorn/prefer-dom-node-dataset": "error",
  "unicorn/prefer-dom-node-remove": "error",
  "unicorn/prefer-dom-node-text-content": "error",
  "unicorn/prefer-includes": "off",
  "unicorn/prefer-keyboard-event-key": "error",
  "unicorn/prefer-math-trunc": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-module": "off",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-protocol": "off",
  "unicorn/prefer-number-properties": Array [
    "warn",
    Object {
      "checkInfinity": false,
    },
  ],
  "unicorn/prefer-object-from-entries": "error",
  "unicorn/prefer-object-has-own": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-prototype-methods": "warn",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-regexp-test": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-string-replace-all": "off",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-string-starts-ends-with": "off",
  "unicorn/prefer-string-trim-start-end": "error",
  "unicorn/prefer-switch": Array [
    "error",
    Object {
      "emptyDefaultCase": "no-default-comment",
    },
  ],
  "unicorn/prefer-ternary": "off",
  "unicorn/prefer-top-level-await": "warn",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/require-array-join-separator": "warn",
  "unicorn/require-number-to-fixed-digits-argument": "warn",
  "unicorn/require-post-message-target-origin": "error",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;

exports[`createUnicornRules matches snapshot without typescript & react 1`] = `
Object {
  "unicorn/better-regex": "error",
  "unicorn/catch-error-name": "error",
  "unicorn/consistent-destructuring": "error",
  "unicorn/consistent-function-scoping": "error",
  "unicorn/custom-error-definition": "warn",
  "unicorn/empty-brace-spaces": "off",
  "unicorn/error-message": "error",
  "unicorn/escape-case": "error",
  "unicorn/expiring-todo-comments": "off",
  "unicorn/explicit-length-check": "error",
  "unicorn/filename-case": "off",
  "unicorn/import-index": "error",
  "unicorn/import-style": "off",
  "unicorn/new-for-builtins": "error",
  "unicorn/no-abusive-eslint-disable": "error",
  "unicorn/no-array-callback-reference": "off",
  "unicorn/no-array-for-each": "off",
  "unicorn/no-array-method-this-argument": "warn",
  "unicorn/no-array-push-push": "warn",
  "unicorn/no-array-reduce": "off",
  "unicorn/no-console-spaces": "error",
  "unicorn/no-document-cookie": "warn",
  "unicorn/no-for-loop": "warn",
  "unicorn/no-hex-escape": "error",
  "unicorn/no-instanceof-array": "error",
  "unicorn/no-keyword-prefix": "warn",
  "unicorn/no-lonely-if": "warn",
  "unicorn/no-nested-ternary": "off",
  "unicorn/no-new-array": "error",
  "unicorn/no-new-buffer": "error",
  "unicorn/no-null": "off",
  "unicorn/no-object-as-default-parameter": "warn",
  "unicorn/no-process-exit": "error",
  "unicorn/no-static-only-class": "error",
  "unicorn/no-this-assignment": "warn",
  "unicorn/no-unreadable-array-destructuring": "warn",
  "unicorn/no-unsafe-regex": "error",
  "unicorn/no-unused-properties": "warn",
  "unicorn/no-useless-length-check": "error",
  "unicorn/no-useless-spread": "error",
  "unicorn/no-useless-undefined": "error",
  "unicorn/no-zero-fractions": "error",
  "unicorn/number-literal-case": "off",
  "unicorn/numeric-separators-style": "warn",
  "unicorn/prefer-add-event-listener": "error",
  "unicorn/prefer-array-find": "error",
  "unicorn/prefer-array-flat": "warn",
  "unicorn/prefer-array-flat-map": "error",
  "unicorn/prefer-array-index-of": "warn",
  "unicorn/prefer-array-some": "warn",
  "unicorn/prefer-at": "off",
  "unicorn/prefer-date-now": "warn",
  "unicorn/prefer-default-parameters": "warn",
  "unicorn/prefer-dom-node-append": "error",
  "unicorn/prefer-dom-node-dataset": "error",
  "unicorn/prefer-dom-node-remove": "error",
  "unicorn/prefer-dom-node-text-content": "error",
  "unicorn/prefer-includes": "warn",
  "unicorn/prefer-keyboard-event-key": "error",
  "unicorn/prefer-math-trunc": "warn",
  "unicorn/prefer-modern-dom-apis": "error",
  "unicorn/prefer-module": "off",
  "unicorn/prefer-negative-index": "error",
  "unicorn/prefer-node-protocol": "off",
  "unicorn/prefer-number-properties": Array [
    "warn",
    Object {
      "checkInfinity": false,
    },
  ],
  "unicorn/prefer-object-from-entries": "error",
  "unicorn/prefer-object-has-own": "warn",
  "unicorn/prefer-optional-catch-binding": "error",
  "unicorn/prefer-prototype-methods": "warn",
  "unicorn/prefer-query-selector": "error",
  "unicorn/prefer-reflect-apply": "error",
  "unicorn/prefer-regexp-test": "error",
  "unicorn/prefer-set-has": "error",
  "unicorn/prefer-spread": "error",
  "unicorn/prefer-string-replace-all": "off",
  "unicorn/prefer-string-slice": "error",
  "unicorn/prefer-string-starts-ends-with": "error",
  "unicorn/prefer-string-trim-start-end": "error",
  "unicorn/prefer-switch": Array [
    "error",
    Object {
      "emptyDefaultCase": "no-default-comment",
    },
  ],
  "unicorn/prefer-ternary": "off",
  "unicorn/prefer-top-level-await": "warn",
  "unicorn/prefer-type-error": "error",
  "unicorn/prevent-abbreviations": "off",
  "unicorn/require-array-join-separator": "warn",
  "unicorn/require-number-to-fixed-digits-argument": "warn",
  "unicorn/require-post-message-target-origin": "error",
  "unicorn/string-content": "off",
  "unicorn/throw-new-error": "error",
}
`;
